### PR TITLE
Negative dim for aten::cat & cat empty tensors.

### PR DESCRIPTION
**Pull request by ailzhang**
`aten::cat` supports negative dim and cat empty tensors. 
This PR fixes the following minimal repro script:
```
import torch
import torch_xla
import torch_xla_py.utils as xu
import torch_xla_py.xla_model as xm

a = torch.Tensor() # EMPTY tensor!

b = torch.rand(2, 2)
xla_device = xm.xla_device()

a = a.to(xla_device)
b = b.to(xla_device)
e = torch.cat([a, b, a], -1) # Negative index!

print(e)
```
I'm not fully familiar with all available APIs in the codebase yet, feel free to let me know if there is a better way to fix it. 
Thanks!

### CODE REVIEW

File: torch_xla/csrc/tensor.cpp [hunk 1]
```diff
@@ -1114,9 +1114,18 @@ XLATensor XLATensor::cat(tensorflow::gtl::ArraySlice<const XLATensor> tensors,
                          xla::int64 dim) {
   XLA_CHECK_GT(tensors.size(), 0);
   std::vector<ir::Value> values;
+
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.cpp]: Remove empty line.

File: torch_xla/csrc/tensor.cpp [hunk 2]
```diff
@@ -1114,9 +1114,18 @@ XLATensor XLATensor::cat(tensorflow::gtl::ArraySlice<const XLATensor> tensors,
...
+  bool set_dim = false;
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.cpp]: I'd prefer simpler code and not keeping the flag around.
When dim turns positive, further calls to GetCanonicalDimension() are just noops.

File: torch_xla/csrc/tensor.cpp [hunk 3]
```diff
@@ -1114,9 +1114,18 @@ XLATensor XLATensor::cat(tensorflow::gtl::ArraySlice<const XLATensor> tensors,
...
-    values.push_back(tensor.GetIrValue());
+    if (xla::ShapeUtil::ElementsIn(tensor.shape()) > 0) {
+      if (!set_dim) {
+        dim = GetCanonicalDimension(tensor, dim);
+        set_dim = true;
+      }
+      values.push_back(tensor.GetIrValue());
+    }
   }
+
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.cpp]: Save vertical space for dire times ;)
